.github/workflows/ci.yml

```diff
@@ -0,0 +1,77 @@
+name: ci
+on:
+  pull_request:
+  push:
+    branches:
+    - master
+  schedule:
+  - cron: '00 01 * * *'
+jobs:
+  test:
+    name: test
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        build:
+        - stable
+        - beta
+        - nightly
+        - macos
+        - win-msvc
+        - win-gnu
+        include:
+        - build: stable
+          os: ubuntu-18.04
+          rust: stable
+        - build: beta
+          os: ubuntu-18.04
+          rust: beta
+        - build: nightly
+          os: ubuntu-18.04
+          rust: nightly
+        - build: macos
+          os: macos-latest
+          rust: stable
+        - build: win-msvc
+          os: windows-2019
+          rust: stable
+        - build: win-gnu
+          os: windows-2019
+          rust: stable-x86_64-gnu
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v1
+      with:
+        fetch-depth: 1
+    - name: Install Rust
+      uses: actions-rs/toolchain@v1
+      with:
+        toolchain: ${{ matrix.rust }}
+        override: true
+        profile: minimal
+    - run: cargo build --verbose
+    - run: cargo doc --verbose
+    - run: cargo test --verbose
+    - run: cargo build --verbose --manifest-path kodama-capi/Cargo.toml
+    - run: cargo build --verbose --manifest-path kodama-bin/Cargo.toml
+    - if: matrix.rust == 'nightly'
+      run: cargo bench --verbose --no-run
+
+  rustfmt:
+    name: rustfmt
+    runs-on: ubuntu-18.04
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v1
+      with:
+        fetch-depth: 1
+    - name: Install Rust
+      uses: actions-rs/toolchain@v1
+      with:
+        toolchain: stable
+        override: true
+        profile: minimal
+        components: rustfmt
+    - name: Check formatting (root)
+      run: |
+        cargo fmt --all -- --check
```

Comment:
Do we additionally want to run something like `clippy` in this section, or another section?

Reply:
I... don't really agree with many of clippy's suggestions and have historically not run it. I'm not alone in that either. Clippy is designed to be opinionated. You can pick and choose which lints you want to run, and I'm not necessarily opposed to doing that, but I think that's well outside the scope of this PR as it would take a fair bit of time.